core/src/main/clojure/xtdb/operator/scan.clj

```diff
@@ -235,7 +235,8 @@
 
 (defprotocol MergePlanPage
   (load-page [mpg buffer-pool vsr-cache])
-  (test-metadata [msg]))
+  (test-metadata [msg])
+  (temporal-bounds [msg]))
```

Comment:
I added a quite generic `temporal-bounds` method. We could add more methods that are finer (`min-sys-from`, `max-sys-to` ....), but the temporal-bounds seemed the best suited.

Reply:
I'm not 100% on this one - it seems like we're re-using TemporalBounds for something it wasn't intended for. That is, originally this was intended for the temporal constraints of a query (and the methods in there assume that); now, we might still have lower and upper on valid-from/sys-from etc but they've got subtly different meanings, so the assumptions of the original methods may be invalid.

For the cost of a new (albeit similar) class, I think separating these out might make it less error-prone?

Reply:
Should mostly be resolved by https://github.com/xtdb/xtdb/pull/3583.